src/App.tsx

```
@@ -7,38 +7,105 @@ import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { client } from './utils/fetchClient';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [userPosts, setUserPosts] = useState<Post[] | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [arePostsLoading, setArePostsLoading] = useState(false);
+  const [hasPostsLoadingError, setHasPostsLoadingError] = useState(false);
+
+  const loadUsers = async () => {
+    try {
+      const loadedUsers: User[] = await client.get('/users');
+
+      setUsers(loadedUsers);
+    } catch {
+      // eslint-disable-next-line no-console
+      console.warn('Could not load users from server');
```

It is better to throw an error instead